mParticle-OneTrust/MPKitOneTrust.m

```diff
@@ -55,6 +73,64 @@ - (void)start {
     });
 }
 
+// Listener for Consent Change dispatched by One Trust
+-(void)actionConsent:(NSNotification*)notification {
+    NSString *category = notification.name; // Cookie Name
+    NSNumber *status = notification.object; // BOOL-ish value for consent
+    category = notification.name;
+
+    // Fire consent change event
+    [self createConsentEvent:category :status];
+}
+
+// Parses the raw consent mapping from the mParticle UI into simple map
+- (NSDictionary*)parseConsentMapping:(NSString*)rawConsentMapping {
+    NSData *jsonData = [rawConsentMapping dataUsingEncoding:NSUTF8StringEncoding];
+    NSError *error;
+
+    NSMutableDictionary *consentMapping = [[NSMutableDictionary alloc] init];
+
+    id jsonObject = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&error];
+
+        if (error) {
+            NSLog(@"Error parsing JSON: %@", error);
+        }
+        else
+        {
+            if ([jsonObject isKindOfClass:[NSArray class]])
+            {
+                for (NSDictionary *element in jsonObject) {
+                    consentMapping[element[@"value"]] = element[@"map"];
+                }
+            }
+            // TODO: Should we throw an error if the consent mapping
+            //       is not an array?
```

Comment:
We should log something like
        NSLog(@"Warning: One Trust Integration initialized with invalid Consent Mapping.\n
        jsonDictionary - %@",jsonDictionary);